### ISSUE TITLE
R Code: Excluding retired mayors

### ISSUE THREAD

**Issue by jmarkgraf** [mberneaud/FinalProject#1]
Hi Malte,

so far I used a rather blunt coding to exclude retired mayors:

```
##: retirement dummy
MayorElection <- slide(MayorElection, Var = "NameCandidate1", 
                       GroupVar = "IDMunicipality",
                       NewVar = "L.NameCandidate1")
MayorElection$Retired <- 1
MayorElection$Retired[MayorElection$NameCandidate1 == 
                        MayorElection$L.NameCandidate1] <- 0
MayorElection$Retired[MayorElection$NameCandidate2 == 
                        MayorElection$L.NameCandidate1] <- 0
MayorElection$Retired[MayorElection$NameCandidate3 == 
                        MayorElection$L.NameCandidate1] <- 0
MayorElection$Retired[MayorElection$NameCandidate4 == 
                        MayorElection$L.NameCandidate1] <- 0
```

To make it really effective, you would indeed need to get rid of the "Dr" titles for all candidate columns. For the moment, it however did the job (although not perfectly). You should find a very strong re-election rate afterwards and a non-significant board-membership dummy. This is at least what I found!

**Comment by mberneaud** [comment]
I used the following code, which should in theory allow for the inclusion of all the different titles as it attempts partial matching of the names if no exact match is found: 

``` R
# creating empty variable to be filled in next step
MayorElection$StandAgain <- NA
# Loop starts at 2nd observation because otherwise 0 would be returned for i-1, 
# which results in NA for the logical condition of the if structure
for(i in 2:nrow(MayorElection)) {
  # Creating a vector with the names of all candidates which ran in it
  allCandidates <- as.character(as.vector(MayorElection[i, c(1, otherCandidateColumns)]))
# Partially matching the name of the previous mayor among the names of all
# candidates of the current election, if it there's a match, 1 will be assigned to
# the StandAgain variable. If there is no match, 0 is assigned.
  if(any(pmatch(MayorElection$NameCandidate1[i-1], allCandidates), na.rm = TRUE)) {
    MayorElection$StandAgain[i] <- 1
  } else {
    MayorElection$StandAgain[i] <- 0
  }
}
```

This works in theory, as it should also match partial strings, but doesn't seem to work correctly, when I tested it by hand. E.g. the code below returns NA if there are any strings in 

``` R
# this does not work
> pmatch("Götz, Franz", "Götz Dr., Franz")
[1] NA

# but this works
> pmatch("Götz, Franz", "Götz, Franz")
[1] 1
```

Which makes it seem like I have to loop the code used for cleaning all the names of the elected candidates over all the remaning columns which contain the names, but that isn't that much work, really. I'll report if the algorithm works as intended afterwards.

**Comment by mberneaud** [comment]
I saw from your code snippet that you changed the names of the other three Columns containing names of candidates, could you share this to make your code work? I manually changed the names for the columns into the ones they are in the data set (all named "Name 2Nachname Titel Vorname" and similar), but no values for the Retire Variable are changed, indicating that the strings in the columns do not match for some reason, indicating that you processed them in a way differently from me. 

Running my code with all the columns cleaned returns 7491 cases where last year's mayor ran again, but I just realised that this code is not applied grouped by municipality, so I have to figure out a way how to do this.

**Comment by jmarkgraf** [comment]
For my analysis I pre-cleaned the xlsx file in Excel - among other steps I changed the variable names before importing the file into R. So, unfortunately there is no code that I can share with you that does that.
Maybe check the string classes that you try to match.

**Comment by mberneaud** [comment]
I had the really strange problem where sourcing my code crashed my R session, which took some re-arranging of the code to make it magically work. 

I think I've found a decently robust way of finding mayors who did not run again combining parts of your method and my method.

**Comment by jmarkgraf** [comment]
Sounds good!

**Comment by mberneaud** [comment]
However, the downside is that the coefficient estimates for Sparkassen board membership turn out insignificant using either method for coding the retired mayors and excluding those who retired. 

Maybe it's a feasible work-around to just include the retired dummy as a control variable, although I feel that this is methodologically not completely sound.

**Comment by jmarkgraf** [comment]
This is what I expected and what I noted in my mail from Sunday. I think this is not a problem per se (at least not for my project). As a next step I would try to run an RD model to see if the smaller number of cases get a coefficient in the ballpark of what Freier (2016) reports...

**Comment by mberneaud** [comment]
RD meaning regression discontinuity design, in this case?

**Comment by jmarkgraf** [comment]
Yes, exactly. This refers to the paper that I cite all the time in my conceptual part by Ronny Freier, who analyzed the incumbency advantage in Bavaria. Not sure if you've read it.

**Comment by jmarkgraf** [comment]
You can also leave out the step of excluding retired mayors in case you would like to have a more exciting finding. For me, the mark does not count and therefore my incentives are different. So, I leave this up to you!

**Comment by mberneaud** [comment]
I ran the logistic regression without excluding (likely) retired mayors, which indeed returns much more interesting results (see below)

![screenshot from 2016-05-11 21-28-26](https://cloud.githubusercontent.com/assets/15064942/15195036/4417fc50-17c6-11e6-9717-9e5bc791822b.png)
However, not accounting for the fact that some mayors do not get reelected because they choose not to run again certainly biases our estimates by counting retiring as failure to become re-elected. On the other hand, one could argue that "losing mayors" are likely underrepresented in the sample, as mayors strategically decide not to run again if they expect not to win. 
Overall, I think it's a bit of a toss-up which effect takes precedence, but I would guess it's the former.

**Comment by jmarkgraf** [comment]
Ok, I'm fine with both options. Just do it as you want. I think the non-significant previous vote share, the direction of the gender effect and the large economic significance of Sparkassen membership make the regression suspect, but maybe that's alright for this context. Could you maybe quickly post the regression output for the sub-sample (excl. retired mayors)? Would be interesting to see how the coefficients/odds change.
Regarding your reasoning for including them: I agree that we might underrepresent the losing mayors, but I would argue that overly ambitious mayors, for instance, that use the mayor position just as a step for a political career on a higher level might make up for that.

**Comment by mberneaud** [comment]
Yeah, I decided to include both sets of regression models into the analysis now, just to get an understanding of the bias we might incur by not controlling for retired mayors, but I made clear that the results are highly suspect. 

I'll include your reasoning as to why losing mayors might not actually be underrepresented into my argument, as it can only strengthen the case of controlling for retired mayors. 

![model1](https://cloud.githubusercontent.com/assets/15064942/15208305/1dcaa5d8-182b-11e6-84d3-291852257562.png)

The above are the results while accounting for mayors who most likely retired. Unfortunately, none of the effects come out to be statistically significant, except for the previous vote share. However, given the small sample size and the small size of the estimated effect of board membership, it is not unlikely to not find any significant effects at this stage. Moreover, there is a strong indication of an incumbency advantage as captured by the previous vote share, which seems more credible than no effect at all. 

Overall, I the model included in this post is a lot more credible and forms the basis for my conclusion, but it is a bit disappointing nevertheless.
